mpas_analysis/test/test_date.py

```diff
@@ -118,5 +119,29 @@ def test_date(self):
         diff = date1-date2
         self.assertEqual(diff, Date(dateString='1995-12-26', isInterval=False))
 
+        date = Date(dateString='1996-01-15', isInterval=False)
+        datetime1 = date.to_datetime(yearOffset=0)
+        datetime2 = datetime.datetime(year=1996, month=1, day=15)
+        self.assertEqual(datetime1, datetime2)
+
+        date = Date(dateString='0000-00-20', isInterval=True)
+        timedelta1 = date.to_timedelta()
+        timedelta2 = datetime.timedelta(days=20)
+        self.assertEqual(timedelta1, timedelta2)
+
+        date = Date(dateString='0001-01-01', isInterval=False)
+        datetime1 = date.to_datetime(yearOffset=0)
+        datetime2 = datetime.datetime(year=1678, month=1, day=1)
+        self.assertEqual(datetime1, datetime2)
+
+        date = Date(dateString='0001-01-01', isInterval=False)
+        datetime1 = date.to_datetime(yearOffset=1849)
+        datetime2 = datetime.datetime(year=1850, month=1, day=1)
+        self.assertEqual(datetime1, datetime2)
+
+        date = Date(dateString='9999-01-01', isInterval=False)
```

Comment:
This tests the clipping capability, right? I think we should note this here and if there is a warning thrown we should make sure it is what we expect to get.